HeterogeneousCore/AlpakaInterface/test/alpaka/testPrefixScan.dev.cc

```diff
@@ -34,7 +34,17 @@ template <typename T>
 struct testPrefixScan {
   template <typename TAcc>
   ALPAKA_FN_ACC void operator()(const TAcc& acc, unsigned int size) const {
+    // alpaka::warp::getSize(acc) is runtime, but we need a compile-time or constexpr value
+#if defined(__CUDA_ARCH__)
+    // CUDA always has a warp size of 32
     auto& ws = alpaka::declareSharedVar<T[32], __COUNTER__>(acc);
+#elif defined(__HIP_DEVICE_COMPILE__)
+    // HIP/ROCm defines warpSize as a constant expression with value 32 or 64 depending on the target device
+    auto& ws = alpaka::declareSharedVar<T[warpSize], __COUNTER__>(acc);
+#else
+    // CPU back-ends always have a warp size of 1
+    auto& ws = alpaka::declareSharedVar<T[1], __COUNTER__>(acc);
+#endif
```

Comment:
Should we think of an abstraction for compile-time defined warp size?

I suppose alternatively these could use the dynamic shared memory.

Reply:
Yes, it's something we are working on.
The main problem is that SYCL allows for different warp sizes, and the actual value is only known at _link_ time (when device compilation happens), not at compile time. 
We have a partially working solution, but it fails in some corner cases, and are discussing with Intel how to fix them. 
Once that is solved, we plan to implement a compile-time constant for the warp size in Alpaka.